[Bug]: Build Failure: Hardcoded ADDON_DIR for matrix-sdk-crypto breaks on newer pnpm versions

### Bug type

Behavior bug (incorrect output/state without crash)

### Beta release blocker

No

### Summary

During the Docker image build process, the native addon verification step fails when using newer versions of pnpm (specifically tested on v10.32.1). The build crashes with:
ERROR: matrix-sdk-crypto native addon missing (pnpm install may have silently failed on this arch)

### Steps to reproduce

Trigger a standard docker build using the repository's Dockerfile.

The build environment uses pnpm v10+.

### Expected behavior

False positive error should not be logged.

### Actual behavior

The build crashes at the verification step because pnpm hashes the virtual store directory differently than older versions.

### OpenClaw version

2026.4.11

### Operating system

Ubuntu 24.04 LTS

### Install method

docker

### Model

any

### Provider / routing chain

n/a

### Additional provider/model setup details

_No response_

### Logs, screenshots, and evidence

```shell
.../@matrix-org/matrix-sdk-crypto-nodejs postinstall: Download Completed
.../@matrix-org/matrix-sdk-crypto-nodejs postinstall: Done
...

==> Verifying critical native addons...
ERROR: matrix-sdk-crypto native addon missing (pnpm install may have silently failed on this arch)
Error: building at STEP "RUN git clone --depth 1 --single-branch -b v2026.4.11         https://github.com/openclaw/openclaw.git /opt/openclaw &&     cd /opt/openclaw &&     pnpm config set registry ${NPM_REGISTRY} &&     pnpm install &&     pnpm build &&     pnpm ui:build &&     echo "==> Verifying critical native addons..." &&     ADDON_DIR=/opt/openclaw/node_modules/.pnpm/@matrix-org+matrix-sdk-crypto-nodejs@0.4.0/node_modules/@matrix-org/matrix-sdk-crypto-nodejs &&     ls "$ADDON_DIR"/matrix-sdk-crypto.linux-*.node 2>/dev/null | grep -q .         || (echo "ERROR: matrix-sdk-crypto native addon missing (pnpm install may have silently failed on this arch)" && exit 1) &&     echo "==> Native addon check passed: $(ls $ADDON_DIR/matrix-sdk-crypto.linux-*.node)" &&     rm -rf         .git .github         test         docker-compose.yml Dockerfile*         scripts/git-hooks         assets README-header.png         .secrets.baseline .pre-commit-config.yaml         .dockerignore .gitignore .gitattributes         .o
make: *** [Makefile:124: build-openclaw-base] Error 1
```

### Impact and severity

Low, workarounds available.

### Additional information

Root Cause
In the Dockerfile, the path to the native addon is hardcoded:

Bash
ADDON_DIR=/opt/openclaw/node_modules/.pnpm/@matrix-org+matrix-sdk-crypto-nodejs@0.4.0/node_modules/@matrix-org/matrix-sdk-crypto-nodejs
Because modern pnpm handles peer-dependency hashing differently, the actual folder name in .pnpm/ gets a hash appended to it (e.g., @matrix-org+matrix-sdk-crypto-nodejs@0.4.0_peerDepHash...), causing the ls check to fail even though the binary successfully downloaded.

Proposed Solution
Replace the brittle hardcoded path with a resilient find command.
Instead of checking a specific .pnpm folder, the Dockerfile should just verify the .node file exists anywhere in the node_modules tree:

Bash
echo "==> Verifying critical native addons..." && \
find /opt/openclaw/node_modules -name "matrix-sdk-crypto.*.node" | grep -q . || \
(echo "ERROR: matrix-sdk-crypto native addon missing" && exit 1)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -26,6 +26,7 @@ Docs: https://docs.openclaw.ai
 - CLI/configure: re-read the persisted config hash after writes so config updates stop failing with stale-hash races. (#64188, #66528)
 - CLI/update: prune stale packaged `dist` chunks after npm upgrades and keep downgrade/verify inventory checks compat-safe so global upgrades stop failing on stale chunk imports. (#66959) Thanks @obviyus.
 - Onboarding/CLI: fix channel-selection crashes on globally installed CLI setups during onboarding. (#66736)
+- Video generation/live tests: bound provider polling for live video smoke, default to the fast non-FAL text-to-video path, and use a one-second lobster prompt so release validation no longer waits indefinitely on slow provider queues.
 - Memory-core/QMD `memory_get`: reject reads of arbitrary workspace markdown paths and only allow canonical memory files (`MEMORY.md`, `memory.md`, `DREAMS.md`, `dreams.md`, `memory/**`) plus exact paths of active indexed QMD workspace documents, so the QMD memory backend can no longer be used as a generic workspace-file read shim that bypasses `read` tool-policy denials. (#66026) Thanks @eleqtrizit.
 - Cron/agents: forward embedded-run tool policy and internal event params into the attempt layer so `--tools` allowlists, cron-owned message-tool suppression, explicit message targeting, and command-path internal events all take effect at runtime again. (#62675) Thanks @hexsprite.
 - Setup/providers: guard preferred-provider lookup during setup so malformed plugin metadata with a missing provider id no longer crashes the wizard with `Cannot read properties of undefined (reading 'trim')`. (#66649) Thanks @Tianworld.
@@ -64,6 +65,7 @@ Docs: https://docs.openclaw.ai
 - Control UI/chat: keep optimistic user message cards visible during active sends by deferring same-session history reloads until the active run ends, including aborted and errored runs. (#66997) Thanks @scotthuang and @vincentkoc.
 - Media/Slack: allow host-local CSV and Markdown uploads only when the fallback buffer actually decodes as text, so real plain-text files work without letting opaque non-text blobs renamed to `.csv` or `.md` slip past the host-read guard. (#67047) Thanks @Unayung.
 - Ollama/onboarding: split setup into `Cloud + Local`, `Cloud only`, and `Local only`, support direct `OLLAMA_API_KEY` cloud setup without a local daemon, and keep Ollama web search on the local-host path. (#67005) Thanks @obviyus.
+- Docker/build: verify `@matrix-org/matrix-sdk-crypto-nodejs` native bindings with `find` under `node_modules` instead of a hardcoded `.pnpm/...` path so pnpm v10+ virtual-store layouts no longer fail the image build. (#67143) thanks @ly85206559.
 
 ## 2026.4.14
 

--- a/Dockerfile
+++ b/Dockerfile
@@ -74,6 +74,12 @@ COPY --from=ext-deps /out/ ./${OPENCLAW_BUNDLED_PLUGIN_DIR}/
 RUN --mount=type=cache,id=openclaw-pnpm-store,target=/root/.local/share/pnpm/store,sharing=locked \
     NODE_OPTIONS=--max-old-space-size=2048 pnpm install --frozen-lockfile
 
+# pnpm v10+ may append peer-resolution hashes to virtual-store folder names; do not hardcode `.pnpm/...`
+# paths. Fail fast here if the Matrix native binding did not materialize after install.
+RUN echo "==> Verifying critical native addons..." && \
+    find /app/node_modules -name "matrix-sdk-crypto*.node" 2>/dev/null | grep -q . || \
+    (echo "ERROR: matrix-sdk-crypto native addon missing (pnpm install may have silently failed on this arch)" >&2 && exit 1)
+
 COPY . .
 
 # Normalize extension paths now so runtime COPY preserves safe modes

--- a/src/dockerfile.test.ts
+++ b/src/dockerfile.test.ts
@@ -42,6 +42,15 @@ describe("Dockerfile", () => {
     expect(dockerfile).toContain("apt-get install -y --no-install-recommends xvfb");
   });
 
+  it("verifies matrix-sdk-crypto native addons without hardcoded pnpm virtual-store paths", async () => {
+    const dockerfile = await readFile(dockerfilePath, "utf8");
+    expect(dockerfile).toContain("Verifying critical native addons");
+    expect(dockerfile).toContain('find /app/node_modules -name "matrix-sdk-crypto*.node"');
+    expect(dockerfile).not.toMatch(
+      /ADDON_DIR=.*node_modules\/\.pnpm\/@matrix-org\+matrix-sdk-crypto-nodejs@/,
+    );
+  });
+
   it("prunes runtime dependencies after the build stage", async () => {
     const dockerfile = await readFile(dockerfilePath, "utf8");
     expect(dockerfile).toContain("FROM build AS runtime-assets");

